ArticleTemplates/assets/js/modules/youtube.js

```diff
@@ -0,0 +1,246 @@
+/*global window,define,YT */
+
+define(function() {
+    'use strict';
+
+    var videos,
+        stateHandlers = {},
+        players = {},
+        progressTracker = {},
+        scriptReady = false;
+
+    function ready() {
+        setStateHandlers();
+        checkForVideos();
+    }
+
+    function setStateHandlers() {
+        if (!GU.opts.nativeYoutubeEnabled || GU.opts.nativeYoutubeEnabled !== 'true') {
+            stateHandlers = {
+                'ENDED': onPlayerEnded,
+                'PLAYING': onPlayerPlaying,
+                'PAUSED': onPlayerPaused
+            };
+        }
+    }
+
+    function setProgressTracker(id) {
+        killProgressTracker(true);
+        progressTracker.id = id;
+        progressTracker.tracker = setInterval(recordPlayerProgress.bind(null, id), 1000);
+    }
+
+    function killProgressTracker(force, id) {
+        if (progressTracker.tracker &&
+            (force || id === progressTracker.id)) {
+            clearInterval(progressTracker.tracker);
+            progressTracker = {};
+        }
+    }
+
+    function checkForVideos() {
+        videos = document.body.querySelectorAll('iframe.youtube-media');
+
+        if (videos.length) {
+            if (!scriptReady) {
+                loadScript();
+            } else {
+                initialiseVideos();
+            }
+        }
+    }
+
+    function loadScript() {
+        var scriptElement;
+
+        if (document.getElementById('youtube-script')) {
+            return;
+        }
+
+        scriptElement = document.createElement('script');
+
+        scriptElement.id = 'youtube-script';
+        scriptElement.async = true;
+        scriptElement.src = 'https://www.youtube.com/iframe_api';
+        scriptElement.onload = onScriptLoaded;
+
+        document.body.appendChild(scriptElement);
+    }
+
+    function onScriptLoaded() {
+        window.onYouTubeIframeAPIReady = onYouTubeIframeAPIReady;
+    }
+
+    function onYouTubeIframeAPIReady() {
+        scriptReady = true;
+        initialiseVideos();
+    }
+
+    function initialiseVideos() {
+        var i,
+            video;
+
+        for (i = 0; i < videos.length; i++) {
+            video = videos[i];
+            if (!players[video.id]) {
+                players[video.id] = {
+                    player: setupPlayer(video.id),
+                    iframe: video,
+                    placeholder: getPlaceholder(video),
+                    trackingCalls: []
+                };
+            }
+        }
+    }
+
+    function getPlaceholder(video) {
+        var placeholder = video.parentNode.getElementsByClassName('youtube-media__placeholder')[0],
+            img = placeholder.getElementsByClassName('youtube-media__placeholder__img')[0];
+
+        if (img.getAttribute('style') !== 'background-image: url()') {
+            return placeholder;
+        } else {
+            placeholder.parentNode.removeChild(placeholder);
+            return false;
```

Comment:
@akash I'm going to update this function as I agree getPlaceholder is not totally descriptive of what's going on here.